### PR TITLE
[BugFix] Fix NPE in tablet_updates::erase_expired_versions

### LINKED ISSUE #6046: BE crash when primary key tablet saving meta
### Steps to reproduce the behavior (Required)
primary key table

### Real behavior (Required)
```
*** Aborted at 1652421582 (unix time) try "date -d @1652421582" if you are using GNU date ***
PC: @          0x17e1f7c starrocks::TabletUpdates::_to_updates_pb_unlocked()
*** SIGSEGV (@0x0) received by PID 1105 (TID 0x7f502df2f700) from PID 0; stack trace: ***
    @          0x350a8a2 google::(anonymous namespace)::FailureSignalHandler()
    @     0x7f5097fca630 (unknown)
    @          0x17e1f7c starrocks::TabletUpdates::_to_updates_pb_unlocked()
    @          0x17e2336 starrocks::TabletUpdates::to_updates_pb()
    @          0x17a3868 starrocks::TabletMeta::_save_meta()
    @          0x17a3e2d starrocks::TabletMeta::save_meta()
    @          0x177e5db starrocks::Tablet::save_meta()
    @          0x17cd561 starrocks::TabletUpdates::remove_expired_versions()
    @          0x179764a starrocks::TabletManager::start_trash_sweep()
    @          0x176909e starrocks::StorageEngine::_start_trash_sweep()
    @          0x175b0d9 starrocks::StorageEngine::_garbage_sweeper_thread_callback()
    @          0x4fe7870 execute_native_thread_routine
    @     0x7f5097fc2ea5 start_thread
    @     0x7f50973c796d __clone
    @                0x0 (unknown)
```
### StarRocks version (Required)
2.1.4

### PR DESCRIPTION
**Pull request by decster**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #6046

## Problem Summary(Required) ：

If creation times of the editversions is not strictly increasing, there may be some editversion ptr left NULL, casing BE crash. This PR fixes this bug.

### CODE REVIEW

File: be/src/storage/tablet_updates.cpp [hunk 1]
```diff
@@ -1133,6 +1142,15 @@ Status TabletUpdates::_commit_compaction(std::unique_ptr<CompactionInfo>* pinfo,
     edit_version_info->rowsets.swap(nrs);
     edit_version_info->compaction.swap(*pinfo);
     _edit_version_infos.emplace_back(std::move(edit_version_info));
+    if (_edit_version_infos.size() >= 2) {
```
Comment (Linkerist) [inline be/src/storage/tablet_updates.cpp:1145]: same logic, how about abstract out this code snippet into a function like "check_if_creation_time_valid"?

File: be/src/storage/tablet_updates.cpp [hunk 2]
```diff
@@ -520,6 +520,15 @@ Status TabletUpdates::_rowset_commit_unlocked(int64_t version, const RowsetShare
     edit_version_info->rowsets.swap(nrs);
     edit_version_info->deltas.push_back(rowsetid);
     _edit_version_infos.emplace_back(std::move(edit_version_info));
+    if (_edit_version_infos.size() >= 2) {
+        auto l1 = _edit_version_infos[_edit_version_infos.size() - 2].get();
```
Comment (Linkerist) [inline be/src/storage/tablet_updates.cpp:524]: the "l" of "l1", "l2" mean "latest" ?

File: be/src/storage/tablet_updates.cpp [hunk 3]
```diff
@@ -1332,6 +1350,8 @@ void TabletUpdates::_erase_expired_versions(int64_t expire_time,
     for (int i = 0; i < _apply_version_idx; i++) {
         if (_edit_version_infos[i]->creation_time <= expire_time) {
             expire_list->emplace_back(std::move(_edit_version_infos[i]));
+        } else {
+            break;
```
Comment (Linkerist) [inline be/src/storage/tablet_updates.cpp:1354]: need this print a log just like the two other changes since that they are the same scene.

Comment (decster) [reply be/src/storage/tablet_updates.cpp:1354]: no it's not

File: be/src/storage/tablet_updates.cpp (hunk 2 above)
Comment (decster) [reply be/src/storage/tablet_updates.cpp:524]: last

File: be/src/storage/tablet_updates.cpp (hunk 1 above)
Comment (decster) [reply be/src/storage/tablet_updates.cpp:1145]: The downside of this is LOG will be at the same line, losing information about which function checks this

Comment (Linkerist) [reply be/src/storage/tablet_updates.cpp:1145]: indeed
i mean the check procedure, not the log line
for example:
if (check_if_creation_time_valid()) {
   // log line..
}
but never mind, this implement also ok.

**Review by Astralidea — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Pull request merged**